docker/build.sh

```diff
@@ -40,6 +40,9 @@ fi
 
 rm -rf runtime-secrets
 
+# register qemu binfmt for automatic emulated arm builds
+docker run --rm --privileged multiarch/qemu-user-static:5.2.0-2 --reset -p yes
```

Comment:
Hopefully the `--reset` flag is implemented in a smart way that avoids the possibility of `exec format error` in other currently running docker builds on the same VM, during the fraction of a second that `binfmt_misc` might be unregistered. The [qemu-user-static README](https://github.com/multiarch/qemu-user-static/blob/master/README.md) reads:

> The `--reset` option is implemented at the register script that executes `find /proc/sys/fs/binfmt_misc -type f -name 'qemu-*' -exec sh -c 'echo -1 > {}' \;` to remove binfmt_misc entry files before register the entry.

This description actually sounds worrying: as if a `find` command in a shell script was executed to unregister `binfmt_misc` entries before then re-registering it. In a VM that was busy with docker builds, this could take, say, 100 milliseconds -- plenty of time for "self-inflicted" `exec format error`.

Reply:
Removing the `--reset` flag and testing in on a NUC with balenaOS, I get:
```
root@9a56d9b:~# balena-engine run --rm --privileged multiarch/qemu-user-static:5.2.0-2 -p yes
sh: write error: File exists
Setting /usr/bin/qemu-alpha-static as binfmt interpreter for alpha
Setting /usr/bin/qemu-arm-static as binfmt interpreter for arm
sh: write error: File exists
Setting /usr/bin/qemu-armeb-static as binfmt interpreter for armeb
sh: write error: File exists
Setting /usr/bin/qemu-sparc-static as binfmt interpreter for sparc
sh: write error: File exists
sh: write error: File exists
sh: write error: File exists
Setting /usr/bin/qemu-sparc32plus-static as binfmt interpreter for sparc32plus
Setting /usr/bin/qemu-sparc64-static as binfmt interpreter for sparc64
Setting /usr/bin/qemu-ppc-static as binfmt interpreter for ppc
sh: write error: File exists
Setting /usr/bin/qemu-ppc64-static as binfmt interpreter for ppc64
sh: write error: File exists
Setting /usr/bin/qemu-ppc64le-static as binfmt interpreter for ppc64le
sh: write error: File exists
Setting /usr/bin/qemu-m68k-static as binfmt interpreter for m68k
sh: write error: File exists
Setting /usr/bin/qemu-mips-static as binfmt interpreter for mips
sh: write error: File exists
Setting /usr/bin/qemu-mipsel-static as binfmt interpreter for mipsel
sh: write error: File exists
sh: write error: File exists
Setting /usr/bin/qemu-mipsn32-static as binfmt interpreter for mipsn32
Setting /usr/bin/qemu-mipsn32el-static as binfmt interpreter for mipsn32el
sh: write error: File exists
Setting /usr/bin/qemu-mips64-static as binfmt interpreter for mips64
sh: write error: File exists
Setting /usr/bin/qemu-mips64el-static as binfmt interpreter for mips64el
sh: write error: File exists
Setting /usr/bin/qemu-sh4-static as binfmt interpreter for sh4
sh: write error: File exists
Setting /usr/bin/qemu-sh4eb-static as binfmt interpreter for sh4eb
sh: write error: File exists
Setting /usr/bin/qemu-s390x-static as binfmt interpreter for s390x
sh: write error: File exists
Setting /usr/bin/qemu-aarch64-static as binfmt interpreter for aarch64
sh: write error: File exists
sh: write error: File exists
Setting /usr/bin/qemu-aarch64_be-static as binfmt interpreter for aarch64_be
sh: write error: File exists
Setting /usr/bin/qemu-hppa-static as binfmt interpreter for hppa
Setting /usr/bin/qemu-riscv32-static as binfmt interpreter for riscv32
sh: write error: File exists
sh: write error: File exists
Setting /usr/bin/qemu-riscv64-static as binfmt interpreter for riscv64
sh: write error: File exists
Setting /usr/bin/qemu-xtensa-static as binfmt interpreter for xtensa
Setting /usr/bin/qemu-xtensaeb-static as binfmt interpreter for xtensaeb
sh: write error: File exists
Setting /usr/bin/qemu-microblaze-static as binfmt interpreter for microblaze
sh: write error: File exists
Setting /usr/bin/qemu-microblazeel-static as binfmt interpreter for microblazeel
sh: write error: File exists
sh: write error: File exists
Setting /usr/bin/qemu-or1k-static as binfmt interpreter for or1k

root@9a56d9b:~# echo $?
0
```
So the exit code is still 0, even if a lot of `File exists` error messages are printed. And it would make it trickier to update the version of  `qemu-user-static` on a running machine. But it may still be preferable to the possibility of a rare `exec format error`... Perhaps adding a comment to the source code:
```
# register qemu binfmt for automatic emulated arm builds
# The `--reset` flag is intentionally not used, to avoid a race condition with
# any ongoing builds on the VM (that could result in `exec format error`).
docker run --rm --privileged multiarch/qemu-user-static:5.2.0-2 --reset -p yes
```

Reply:
Sounds good to me!

Reply:
If there was a "worker setup" `balena-concourse` script that got executed when the worker was added... Then we could have `qemu-user-static --reset` executing in that script, instead of at the beginning of each build job.

Reply:
That would be preferable, as long as we could guarantee it would always be called on a new worker and not forgotten or dismissed if/when we move to new workers or new CI. I like it close to where it is needed for now.